golem-worker-bridge/src/oas_worker_bridge.rs

```diff
@@ -0,0 +1,125 @@
+use crate::api_definition::{
+    ApiDefinition, ApiDefinitionId, GolemWorkerBinding, MethodPattern, PathPattern, Route, Version,
+};
+use crate::expr::Expr;
+use golem_common::model::TemplateId;
+use openapiv3::OpenAPI;
+use serde_json;
+use serde_json::Value;
+use uuid::Uuid;
+
+pub fn get_api_definition(open_api: &str) -> Result<ApiDefinition, String> {
+    dbg!(open_api);
+    let openapi: OpenAPI = serde_json::from_str(open_api).map_err(|e| e.to_string())?;
+    let version = Version(openapi.info.version);
+    let id = ApiDefinitionId(
+        openapi
+            .extensions
+            .iter()
+            .find(|(key, _)| key.to_lowercase() == "x-api-definition-id")
```

Comment:
i would do `x-api-definition-id` -> `x-golem-api-definition-id` - if it is golem specific

Reply:
This is very rough PR, I can take reviews soon

Reply:
Will fix in next PR. Right now I got working examples, and didn't want to go through testing it again

Reply:
maybe is good to have constants